### PR TITLE
Add review ratings to ElasticSearch

### PR DESCRIPTION
**Pull request by cewald**
It's not possible to show a star rating in the VSF right now.
But I think this is a common feature and trust signal.

If it is not to much overhead to the review document type I would suggest to add this to the Magento 1.9 indexer by default.

### CODE REVIEW

File: src/app/code/community/Divante/VueStorefrontIndexer/Model/Indexer/Action/Reviews.php [hunk 1]
```diff
@@ -44,11 +44,17 @@ public function rebuild($storeId = 1, array $reviewIds = [])
         do {
             $reviews = $this->resourceModel->getReviews($storeId, $reviewIds, $lastReviewId);
 
+            /** @var Divante_VueStorefrontIndexer_Model_Resource_Catalog_Rating $ratingModel */
+            $ratingsModel = Mage::getResourceModel('icmaa_vuestorefrontindexer/catalog_rating');
```
Comment (ResuBaka) [inline src/app/code/community/Divante/VueStorefrontIndexer/Model/Indexer/Action/Reviews.php]: Can you change that to `vsf_indexer/catalog_rating` as that I think should be the right name for it.

Comment (cewald) [reply src/app/code/community/Divante/VueStorefrontIndexer/Model/Indexer/Action/Reviews.php]: Thanks for noticing – I copy and paste that one from our module.

**Comment by afirlejczyk** [comment]
Hi.

1. I would move getting ratings to datasource
You can check datasource for taxes or product/categories
2. Rating title can be empty (title for specific store view), we should use Rating Code from `rating` table as fallback.
3. `rating_id` - I don't think so it is necessary, but it can stay. ;)

I just was working for ratings for magento2 module, 
https://github.com/DivanteLtd/magento2-vsbridge-indexer/pull/117/files

**Comment by cewald** [comment]
@afirlejczyk I've changed your points. Thanks for the feedback.
I removed `rating_id` – I first guessed that it would be handy if you link stats from a 3d-party system which also relies on Magento's rating results. But, yes thats probably not necessary.

File: src/app/code/community/Divante/VueStorefrontIndexer/Model/Indexer/Datasource/Review/Rating.php [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        $this->resourceModel = Mage::getResourceModel('vsf_indexer/catalog_rating');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function addData(array $indexData, $storeId)
+    {
+        $reviewResultIds = array_column($indexData, 'id');
+        $this->resourceModel->getRatings($storeId, $reviewResultIds);
```
Comment (afirlejczyk) [inline src/app/code/community/Divante/VueStorefrontIndexer/Model/Indexer/Datasource/Review/Rating.php:35]: As we exporting data in batch, we should probably reset reviews in resource model before getting ratings for the next batch.
`Divante_VueStorefrontIndexer_Model_Indexer_Datasource_Review_Rating` - we probably have the same class instance for datasource all the time (for all batches). I think.. I will double check. 

but we can just do
```
$this->resourceModel->clearRatings()
$this->resourceModel->getRatings($storeId, $reviewResultIds);
```

File: src/app/code/community/Divante/VueStorefrontIndexer/Model/Resource/Catalog/Rating.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+                ]
+            );
+    
+            $select->where('e.review_id IN (?)', $reviewIds);
+    
+            $select->joinLeft(
+                ['r' => $this->coreResource->getTableName('rating')],
+                'e.rating_id = r.rating_id',
+                ['title' => 'rating_code']
+            )->order('e.review_id ASC');
```
Comment (afirlejczyk) [inline src/app/code/community/Divante/VueStorefrontIndexer/Model/Resource/Catalog/Rating.php:67]: I would use `value` from `rating_title` table-  if available, and `rating_code` from `rating` table if not.
```
$table = 'r';
        $codeExpr = $connection->getIfNullSql('title.value', "{$table}.rating_code");
            $select->joinLeft(
                ['title' => $this->resource->getTableName('rating_title')],
                $connection->quoteInto("{$table}.rating_id = title.rating_id AND title.store_id = ?", $storeId),
                ['title' => $codeExpr]
            );
```

**Comment by afirlejczyk** [comment]
Hi, Thanks for changes.
I will merge this and make two small adjustments!

**Pull request merged**